src/util/snippets.py

```diff
@@ -243,11 +249,19 @@ def _dumps(self, node):
 
 
 class Convertor:
-    def __init__(self, converters, callback):
+    def __init__(self, converters, callback, dbname, update_query):
         self.converters = converters
         self.callback = callback
+        self.dbname = dbname
+        self.update_query = update_query
+
+    def __call__(self, query):
+        with db_connect(self.dbname).cursor() as cr:
+            cr.execute(query)
+            for changes in filter(None, map(self._convert_row, cr.fetchall())):
+                cr.execute(self.update_query, changes)
```

Comment:
We need to ensure we close the connection. Also I think we are missing a commit for the cursor 🤔 (maybe there is some sort of autocommit logic I'm missing here).
```suggestion
        conn = db_connect(self.dbname)
        updated = False
        with conn, conn.cursor() as cr:
            cr.execute(query)
            for changes in filter(None, map(self._convert_row, cr.fetchall())):
                updated = True
                cr.execute(self.update_query, changes)
            if updated:
                cr.commit()
        conn.close()
```

Reply:
The contextmanager commits and closes the cursor.
https://github.com/odoo/odoo/blob/17.0/odoo/sql_db.py#L164-L172
I do not want to close the connection, here. It will be added to the pool and re-used upon each call to `__call__()`, i.e. each further query assigned to the worker process. The connection socket is destroyed along with the filehandle along with the child process.
I would have liked to do that explicitly, but the `ProcessPoolExecutor` is limited in that it has an api hook/callback only for initialization code,  not for destruction code.